api/api.proto

```diff
@@ -911,6 +911,86 @@ service OpenStorageCredentials {
     }
 }
 
+service OpenStorageSchedulePolicy {
+
+  // Create Schedule Policy for snapshots
+  rpc Create(SdkSchedulePolicyCreateRequest)
+    returns (SdkSchedulePolicyCreateResponse) {
+      option(google.api.http) = {
+        post: "/v1/schedulepolicy/create"
+        body: "*"
+      };
+    }
+
+  // Update Schedule Policy
+  rpc Update(SdkSchedulePolicyUpdateRequest)
+    returns (SdkSchedulePolicyUpdateResponse) {
+      option(google.api.http) = {
+        post: "/v1/schedulepolicy/update"
+        body: "*"
+      };
+    }
+
+  // Enumerate Schedule Policy
+  rpc Enumerate(SdkSchedulePolicyEnumerateRequest)
+    returns (SdkSchedulePolicyEnumerateResponse) {
+      option(google.api.http) = {
+        post: "/v1/schedulepolicy/enumerate"
+        body: "*"
+      };
+    }
+
+  // Delete Schedule Policy
+  rpc Delete(SdkSchedulePolicyDeleteRequest)
+    returns (SdkSchedulePolicyDeleteResponse) {
+      option(google.api.http) = {
+        post: "/v1/schedulepolicy/delete"
+        body: "*"
+      };
+    }
+}
+
+message SdkSchedulePolicyCreateRequest {
+  // Schedule Policy
+  SdkSchedulePolicy SchedulePolicy = 1;
+}
+
+message SdkSchedulePolicyCreateResponse {
+}
+
+message SdkSchedulePolicyUpdateRequest {
+  // Schedule Policy
+  SdkSchedulePolicy SchedulePolicy = 1;
+}
+
+message SdkSchedulePolicyUpdateResponse {
+}
+
+message SdkSchedulePolicyEnumerateRequest {
+  // Name of the schedule Policy
+  string name = 1;
+}
+
+message SdkSchedulePolicyEnumerateResponse {
+  // List of Schedule Policy
+  repeated SdkSchedulePolicy policies = 1;
+}
+
+message SdkSchedulePolicyDeleteRequest {
+  // Name of the schedule policy
+  string name = 1;
+}
+
+message SdkSchedulePolicyDeleteResponse {
+}
+
+message SdkSchedulePolicy {
+  // Name of the schedule policy
+  string name = 1;
+  // Schedule string in yaml
+  string schedule = 2;
```

Comment:
This should probably not be a string. Looks like multiple values are provided encoded here. Not part of this PR, but something that needs to be addressed before release.

Reply:
from what  i recall , this is a yaml , which is passed as a string.

Reply:
Created https://github.com/libopenstorage/openstorage/issues/459 to track